yt/visualization/fixed_resolution.py

```diff
@@ -708,21 +712,40 @@ def __getitem__(self, item):
             deposition,
         )
 
-        bounds = []
-        for b in self.bounds:
-            if hasattr(b, "in_units"):
-                b = float(b.in_units("code_length"))
-            bounds.append(b)
+        dd = self.data_source.dd
 
         ftype = item[0]
-        x_data = self.data_source.dd[ftype, self.x_field]
-        y_data = self.data_source.dd[ftype, self.y_field]
-        data = self.data_source.dd[item]
+        if self.axis is None:
+            wd = []
+            for w in self.data_source.width:
+                if hasattr(w, "to_value"):
+                    w = w.to_value("code_length")
+                wd.append(w)
+            x_data, y_data, *bounds = rotate_particle_coord(
+                dd[ftype, "particle_position_x"].to_value("code_length"),
+                dd[ftype, "particle_position_y"].to_value("code_length"),
+                dd[ftype, "particle_position_z"].to_value("code_length"),
+                self.data_source.center.to_value("code_length"),
+                wd,
+                self.data_source.normal_vector,
+                self.data_source.north_vector,
+            )
+            x_data = np.array(x_data)
+            y_data = np.array(y_data)
+        else:
+            bounds = []
+            for b in self.bounds:
+                if hasattr(b, "to_value"):
+                    b = b.to_value("code_length")
+                bounds.append(b)
+            x_data = dd[ftype, self.x_field].to_value("code_length")
+            y_data = dd[ftype, self.y_field].to_value("code_length")
+        data = dd[item]
 
         # handle periodicity
-        dx = x_data.in_units("code_length").d - bounds[0]
-        dy = y_data.in_units("code_length").d - bounds[2]
-        if self.periodic:
+        dx = x_data - bounds[0]
+        dy = y_data - bounds[2]
+        if self.axis is not None and self.periodic:
```

Comment:
haven't we established that `self.periodic is False` implies `self.axis is None` ? I may have missed something but the extra condition seems superfluous to me

Reply:
Probably superfluous, will have to check.